sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -998,10 +998,27 @@ abstract class BinaryComparison extends BinaryOperator with Predicate {
     }
   }
 
-  protected lazy val ordering: Ordering[Any] = TypeUtils.getInterpretedOrdering(left.dataType)
+  protected lazy val ordering: Ordering[Any] = new Ordering[Any] {
```

Comment:
```suggestion
  protected lazy val ordering: Ordering[Any] = new Ordering[Any] {
  override def compare(x: Any, y: Any): Int = {
    (x, y) match {
      case (xs: String, yi: Int) => xs.compareTo(yi.toString)
      case (xi: Int, ys: String) => xi.toString.compareTo(ys)
      case _ => TypeUtils.getInterpretedOrdering(left.dataType).compare(x, y)
    }
  }
}

```

Reply:
> 

I m not sure if this will help.  @Masykus

Reply:
Thank you for your suggestion! 
I think your solution is more robust as it utilizes compareTo for string comparisons and falls back to TypeUtils.getInterpretedOrdering for other types.

The logic I was about to commit is as below, but I wasn't sure which way would be better
@Masykus Can you please take a look? Thank you!
```suggestion
    override def compare(x: Any, y: Any): Int = {
      (x, y) match {
        case (xs: String, yi: Int) => xs.compare(yi.toString)
        case (xi: Int, ys: String) => xi.toString.compare(ys)
        case _ => super.compare(x, y)
      }
    }
```